peers/peerpool.go

```diff
@@ -132,9 +134,15 @@ func (p *PeerPool) Start(server *p2p.Server) error {
 	}()
 
 	// collect topics and start searching for nodes
-	p.topics = make([]*TopicPool, 0, len(p.config))
+	p.topics = make([]TopicPoolInterface, 0, len(p.config))
 	for topic, limits := range p.config {
-		topicPool := NewTopicPool(p.discovery, topic, limits, p.opts.SlowSync, p.opts.FastSync, p.cache)
+		var topicPool TopicPoolInterface
+		t := newTopicPool(p.discovery, topic, limits, p.opts.SlowSync, p.opts.FastSync, p.cache)
+		if topic == MailServerDiscoveryTopic {
```

Comment:
what i suggested is to construct topic pools and pass them to peer pool constructor instead of config. tbh i see that as much better than condition for particular topic